### PR TITLE
WT-8784 Fix invalid S3CacheExists if condition

### PR DESCRIPTION
**Pull request by fionaochee**

### CODE REVIEW

**Review by etienneptl — APPROVED** [review]
LGTM

File: ext/storage_sources/s3_store/s3_storage_source.cpp [hunk 1]
```diff
@@ -98,7 +98,7 @@ S3Exist(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session, const char *name, bool
     S3_FILE_SYSTEM *fs = (S3_FILE_SYSTEM *)fileSystem;
 
     /* It's not in the cache, try the S3 bucket. */
-    if (*exist = !S3CacheExists(fileSystem, name))
+    if (*exist = S3CacheExists(fileSystem, name) == false)
```
Comment (quchenhao) [inline ext/storage_sources/s3_store/s3_storage_source.cpp:101]: Can you do
```
*exist = S3CacheExists(fileSystem, name);
if (!exist)
```

**Review by jiechenbo — APPROVED** [review]
LGTM, nitpick: I would prefer if a bracket is between (*exist = S3CacheExists(fileSystem, name)) before checking for false.

**Review by quchenhao — APPROVED** [review]
lgtm

**Pull request merged**